### PR TITLE
centralization of notifications

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java [hunk 1]
```diff
@@ -57,8 +51,15 @@ public Mono<UUID> runAndSaveResult(SecurityAnalysisRunContext runContext) {
 
         // update status to running status
         return setStatus(List.of(resultUuid), SecurityAnalysisStatus.RUNNING.name()).then(
-                Mono.fromRunnable(() ->
-                        sendRunMessage(new SecurityAnalysisResultContext(resultUuid, runContext).toMessage(objectMapper)))
+                Mono.fromRunnable(() -> {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java:54]: Why do you remove the `SecurityAnalysisResultContext::toMessage` ?

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java [hunk 2]
```diff
@@ -83,17 +84,6 @@ public Mono<Void> setStatus(List<UUID> resultUuids, String status) {
...
-
-    private void sendRunMessage(Message<String> message) {
-        RUN_MESSAGE_LOGGER.debug("Sending message : {}", message);
-        publisher.send("publishRun-out-0", message);
-    }
-
-    private void sendCancelMessage(Message<String> message) {
-        CANCEL_MESSAGE_LOGGER.debug("Sending message : {}", message);
-        publisher.send("publishCancel-out-0", message);
+        return Mono.fromRunnable(() -> notificationService.emitCancelAnalysisMessage(resultUuid.toString(), receiver)).then();
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java:87]: Same

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java:54]: I create the message directly in NotificationService.

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java [hunk 3]
```diff
@@ -68,22 +71,18 @@ public class SecurityAnalysisWorkerService {
 
     private static final String CATEGORY_BROKER_OUTPUT = SecurityAnalysisWorkerService.class.getName() + ".output-broker-messages";
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java]: ok. done

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java]: Ok. done

File: src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisService.java]: I moved back the creation of the message to SecurityAnalysisResultContext

**Review by bendaamerahm — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**